research/flamby/fed_isic2019/apfl/client.py

```diff
@@ -1,60 +1,76 @@
-from fl4health.parameter_exchange.layer_exchanger import FixedLayerExchanger
-from fl4health.utils.metrics import BalancedAccuracy, Metric
+from fl4health.utils.losses import LossMeterType
+from fl4health.utils.metrics import BalancedAccuracy, Metric, MetricMeterType
 from research.flamby.fed_isic2019.apfl.apfl_model import ApflEfficientNet
-from research.flamby.flamby_clients.flamby_apfl_client import FlambyApflClient
 from research.flamby.flamby_data_utils import construct_fedisic_train_val_datasets
 
 
-class FedIsic2019ApflClient(FlambyApflClient):
+class FedIsic2019ApflClient(ApflClient):
     def __init__(
         self,
-        learning_rate: float,
-        alpha_learning_rate: float,
+        data_path: Path,
         metrics: Sequence[Metric],
         device: torch.device,
         client_number: int,
-        checkpoint_stub: str,
-        dataset_dir: str,
-        run_name: str = "",
+        learning_rate: float,
+        alpha_learning_rate: float,
+        loss_meter_type: LossMeterType = LossMeterType.AVERAGE,
+        metric_meter_type: MetricMeterType = MetricMeterType.ACCUMULATION,
+        checkpointer: Optional[TorchCheckpointer] = None,
     ) -> None:
-        assert 0 <= client_number < NUM_CLIENTS
         super().__init__(
-            learning_rate, alpha_learning_rate, metrics, device, client_number, checkpoint_stub, dataset_dir, run_name
+            data_path=data_path,
+            metrics=metrics,
+            device=device,
+            loss_meter_type=loss_meter_type,
+            metric_meter_type=metric_meter_type,
+            checkpointer=checkpointer,
         )
+        assert 0 <= client_number < NUM_CLIENTS
 
-    def setup_client(self, config: Config) -> None:
-        train_dataset, validation_dataset = construct_fedisic_train_val_datasets(self.client_number, self.dataset_dir)
-
-        self.train_loader = DataLoader(train_dataset, batch_size=BATCH_SIZE, shuffle=True)
-        self.val_loader = DataLoader(validation_dataset, batch_size=BATCH_SIZE, shuffle=False)
+        self.learning_rate = learning_rate
+        self.alpha_learning_rate = alpha_learning_rate
+        self.client_number = client_number
 
-        self.num_examples = {"train_set": len(train_dataset), "validation_set": len(validation_dataset)}
+    def get_data_loaders(self, config: Config) -> Tuple[DataLoader, DataLoader]:
+        train_dataset, validation_dataset = construct_fedisic_train_val_datasets(
+            self.client_number, str(self.data_path)
+        )
+        train_loader = DataLoader(train_dataset, batch_size=BATCH_SIZE, shuffle=True)
+        val_loader = DataLoader(validation_dataset, batch_size=BATCH_SIZE, shuffle=False)
+        return train_loader, val_loader
 
-        # NOTE: The class weights specified by alpha in this baseline loss are precomputed based on the weights of
-        # the pool dataset. This is a bit of cheating but FLamby does it in their paper.
```

Comment:
We lost this comment in the refactor. Do you mind adding it back to `get_criterion` It's just a note about the baseline loss for FedIsic

Reply:
Updated in [9959c89](https://github.com/VectorInstitute/FL4Health/pull/62/commits/9959c896eba8b13846f86a313be23b34619dc1cb)